dali/pipeline/workspace/sample_workspace.h

```diff
@@ -72,6 +72,18 @@ class DLL_PUBLIC SampleWorkspace : public WorkspaceBase<SampleInputType, SampleO
   template <typename Backend>
   DLL_PUBLIC Tensor<Backend>& Output(int idx);
 
+  int GetInputBatchSize(int) const {
```

Comment:
I got mixed feelings. If we don't allow the ops using SampleWorkspace to use them, maybe it's better to not add this? I know I was thinking about making those ops responsible for propagating the BS, but maybe it can be done in the default Setup/RunImpl for HostWorkspace.